### DISCUSSION TITLE
emitAsset not working from within Webpack Plugin

### DISCUSSION THREAD

**Discussion by nmn** [vercel/next.js#32836] [Help]
I'm working on integrating Stylex (Facebook's internal, soon to be open-source, CSS-in-JS library).

The way that stylex works, is that you write your styles in the same file as your components. A static compile step converts style definitions to classNames and a static CSS file is emitted.

In the case of Webpack, we use the `babel-loader` to collect the generated CSS on a `metadataSubscribers` and the use a compilation hook, to `emitAsset` when it's all done.

The problem that I'm running into is that `emitAsset` doesn't seem to do anything when used within Next.JS.

We have tested this same Webpack plugin in a standalone Webpack project, and I have verified that the function *is* being called.

***

As a temporary workaround, I tried to use `fs.writeFileSync` to write the file manually where the `emitAsset` function is called and it seems to work correctly. However this is brittle, and I'm also being forced to add a `link` tag manually to use this generated CSS file.

***

Any help with any of this would be greatly appreciated.

**Comment by sag1v** [comment]
@nmn Did you ever figure out why it didn't work?

I'm using webpack v5 and can't get it to work either.
